### PR TITLE
[DPP-771][Error codes] Add readme.md and supplement docs with a concrete error handling example.

### PR DESCRIPTION
**Pull request by pbatko-da**

### CODE REVIEW

File: docs/source/app-dev/grpc/error-codes.rst [hunk 1]
```diff
@@ -157,6 +157,82 @@ but there is no guarantee given that additional information will be preserved ac
 
 
 
+Working with Error Codes
+======================================
+
+The example below shows how a user can in practice extract relevant error information.
```
Comment (mziolekda) [inline docs/source/app-dev/grpc/error-codes.rst:163]:
```suggestion
This example shows how a user can extract the relevant error information.
```

File: docs/source/app-dev/grpc/error-codes.rst [hunk 2]
```diff
@@ -157,6 +157,82 @@ but there is no guarantee given that additional information will be preserved ac
...
+            assert {
+              rawDetails.collectFirst {
+                case any if any.is(classOf[ErrorInfo]) =>
+                  val v = any.unpack(classOf[ErrorInfo])
+                  assert(v.getReason == "MY_ERROR_CODE_ID")
+                  assert(v.getMetadataMap.asScala.toMap == Map("category" -> "2", "foo" -> "bar"))
+              }.isDefined
+            }
+
+            // Extracting full correlation id if present.
```
Comment (mziolekda) [inline docs/source/app-dev/grpc/error-codes.rst:204]:
```suggestion
            // Extracting full correlation id, if present.
```

File: ledger/error/src/main/scala/com/daml/error/samples/Example.scala [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  def serviceEndpointDummy(): Unit = {
+    throw ErrorCodeFoo.Error("A user oriented message").asGrpcError
+  }
+
+}
+
+/** This shows how a user can handle error codes.
+  * In particular it shows how to extract useful information from the signalled exception with minimal library dependencies.
+  *
+  * NOTE: This class is given as an example in the official Daml documentation. If you change it here change it also in the docs.
```
Comment (mziolekda) [inline ledger/error/src/main/scala/com/daml/error/samples/Example.scala:56]:
```suggestion
  * NOTE: This class is given as an example in the official Daml documentation. If you change it here, change it also in the docs.
```

**Review by mziolekda — APPROVED** [review]
Few minor nit-picks

File: ledger/error/src/main/scala/com/daml/error/samples/Example.scala [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        assert {
+          rawDetails.collectFirst {
+            case any if any.is(classOf[ErrorInfo]) =>
+              val v = any.unpack(classOf[ErrorInfo])
+              assert(v.getReason == "MY_ERROR_CODE_ID")
+              assert(v.getMetadataMap.asScala.toMap == Map("category" -> "2", "foo" -> "bar"))
+          }.isDefined
+        }
+
+        // Extracting full correlation id if present.
```
Comment (pbatko-da) [inline ledger/error/src/main/scala/com/daml/error/samples/Example.scala:94]:
```suggestion
        // Extracting full correlation id, if present.
```

**Pull request merged**